### PR TITLE
Use modern syntax in test-case files where possible

### PR DESCRIPTION
**Pull request by AlexWaygood**
By default, pyupgrade only rewrites `.py` files, so adding it as a pre-commit hook should have no effect on the `stubs/` or `stdlib/` directories.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 repos:
+  - repo: https://github.com/asottile/pyupgrade
```
Comment (AlexWaygood) [inline .pre-commit-config.yaml:2]:
```suggestion
  # this only runs on the .py files
  - repo: https://github.com/asottile/pyupgrade
```

**Comment by AlexWaygood** [comment]
The stubtest failures in `pycurl` are unrelated (presumably we'll have a daily test failure this evening).

**Review by hauntsaninja — COMMENTED** [review]
Not sure we want to run this automatically,e.g. we're changing what we're type checking in the check_post example

**Comment by AlexWaygood** [comment]
> Not sure we want to run this automatically,e.g. we're changing what we're type checking in the check_post example

Yeah, that's fair.

**Pull request merged**